app/controllers/asset/AddAssetsController.scala

```diff
@@ -66,35 +66,39 @@ class AddAssetsController @Inject()(
     }
   }
 
-  def onPageLoad(): Action[AnyContent] = standardActionSets.verifiedForIdentifier {
+  def onPageLoad(): Action[AnyContent] = standardActionSets.verifiedForIdentifier.async {
     implicit request =>
 
       val userAnswers: UserAnswers = request.userAnswers
       val isTaxable = userAnswers.isTaxable
 
-      val assets = new AddAssetViewHelper(userAnswers).rows
-
-      val maxLimit: Int = (userAnswers.is5mldEnabled, isTaxable) match {
-        case (true, true) => MAX_5MLD_TAXABLE_ASSETS
-        case (true, false) => MAX_5MLD_NON_TAXABLE_ASSETS
-        case _ => MAX_4MLD_ASSETS
-      }
+      for {
+        assets <- trustService.getAssets(userAnswers.identifier)
```

Comment:
Don't forget to cleanup user answers here. Happy to merge without though as work in progress

Reply:
Yep. This is my next task, completely updating all this code and re-writing all the tests